### PR TITLE
Reduce use of binders

### PR DESCRIPTION
**Pull request by K-ballo**
Replace several uses of binders with simpler lighter alternatives.

### CODE REVIEW

File: hpx/lcos/dataflow.hpp [hunk 1]
```diff
@@ -193,8 +193,10 @@ namespace hpx { namespace lcos { namespace detail
             // simply schedule new thread
             parallel::execution::parallel_policy_executor<launch::async_policy>
                 exec{policy};
-            parallel::execution::post(exec, &dataflow_frame::done,
-                std::move(this_), std::move(futures));
+            parallel::execution::post(exec,
+                [HPX_CAPTURE_MOVE(this_)](Futures&& futures) mutable -> void {
+                    return this_->done(std::move(futures));
+                }, std::move(futures));
```
Comment (hkaiser) [inline hpx/lcos/dataflow.hpp]: What's the rationale of this change?

Comment (K-ballo) [reply hpx/lcos/dataflow.hpp]: Lifting the mem-ptr value from runtime into compile time, yielding a smaller, simpler, and easier to inline callable.
